hazelcast/src/main/java/com/hazelcast/client/impl/proxy/ClientTopicProxy.java

```diff
@@ -77,6 +95,27 @@ public LocalTopicStats getLocalTopicStats() {
         throw new UnsupportedOperationException("Locality is ambiguous for client!");
     }
 
+    @Override
+    public void publishAll(@Nonnull Collection<? extends E> messages) {
+        checkNotNull(messages, NULL_MESSAGE_IS_NOT_ALLOWED);
+        checkNoNullInside(messages, NULL_MESSAGE_IS_NOT_ALLOWED);
+
+        Collection<Data> dataCollection = objectToDataCollection(messages, getSerializationService());
+
+        ClientMessage request = TopicPublishAllCodec.encodeRequest(name, dataCollection);
+        invokeOnPartition(request);
+    }
+
+    @Override
+    public InternalCompletableFuture<E> publishAllAsync(@Nonnull Collection<? extends E> messages) {
+        checkNotNull(messages, NULL_MESSAGE_IS_NOT_ALLOWED);
+        checkNoNullInside(messages, NULL_MESSAGE_IS_NOT_ALLOWED);
+
+        Collection<Data> dataCollection = objectToDataCollection(messages, getSerializationService());
+        final ClientMessage clientMessage = TopicPublishAllCodec.encodeRequest(name, dataCollection);
+        return publishAsyncInternal(clientMessage);
```

Comment:
Also done. :)

Reply:
With the latest iteration of changes, this would become:
```java
    @Override
    public CompletionStage<Void> publishAsync(@Nonnull E message) {
        checkNotNull(message, NULL_MESSAGE_IS_NOT_ALLOWED);

        Data data = toData(message);
        final ClientMessage clientMessage = TopicPublishCodec.encodeRequest(name, data);
        InternalCompletableFuture<Void> returnFuture = new InternalCompletableFuture<>();
        publishAsyncInternal(clientMessage).whenCompleteAsync((o, t) -> {
            if (t != null) {
                returnFuture.completeExceptionally(t);
            } else {
                returnFuture.complete(null);
            }
        });
        return returnFuture;
    }

    protected ClientInvocationFuture publishAsyncInternal(ClientMessage clientMessage) {
        try {
            return new ClientInvocation(getClient(), clientMessage, getName(), getPartitionId()).invoke();
        } catch (Exception e) {
            throw rethrow(e);
        }
    }
```